cupy/cuda/stream.pyx

```diff
@@ -275,9 +283,11 @@ class Stream(BaseStream):
 
     """
 
-    def __init__(self, null=False, non_blocking=False):
+    def __init__(self, null=False, ptds=False, non_blocking=False):
         if null:
-            self.ptr = 0
+            self.ptr = runtime.cudaStreamDefault
```

Comment:
Technically speaking, the `cudaStreamDefault`/`cudaStreamNonBlocking` are flags for [`cudaStreamCreateWithFlags`](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__STREAM.html#group__CUDART__STREAM_1gb1e32aff9f59119e4d0a9858991c4ad3), but it happens that `cudaStreamDefault` is also `0`, which is consistent with the default behavior for when no streams are passed and is also used by many people, CuPy included. On the other hand `cudaStreamLegacy` and `cudaStreamPerThread` are _stream handles_ and these are the ones that we should be using when specifying a stream explicitly.

Reply:
Right, so this is exactly I was thinking: Don't we expect `self.ptr` to be a *stream handle* and so it should be `runtime. cudaStreamLegacy`?

Reply:
_Probably_. The default was `0`, so I thought I would play it safe and avoid replacing that by `cudaStreamLegacy` for now. I believe the behavior should be the same though. Maybe it's a good time to replace and test that out, unless someone has any reasons not to do so that I may be overlooking.

Reply:
I adjusted this now to `runtime.cudaStreamLegacy`, so far everything looks correct on my verification.